core/meta/inc/TClass.h

```diff
@@ -132,6 +132,26 @@ friend class TStreamerInfo;
                         // in ROOT Meta w/o interpreter information
    };
 
+   // "Typed" pointer that recalls how TClass::New allocated the object.
+   // It is returned by TClass:NewObject and should be passed to TClass::DeleteArray or TClass::Destructor
+   // to delete the object.
+   // It is also used in TVirtualCollectionProxy for the same reasons.
+   class ObjectPtr
+   {
+      void *fPtr = nullptr;
```

Comment:
Should this not be `char *`? Makes e.g. `TVirtualArray::GetObjectAt()` much simpler.

Reply:
It could be but semantic it is a void* so it might be clearer as is, especially if we make it a struct.